app/assets/src/components/common/DetailsSidebar/PipelineStepDetailsMode/PipelineStepDetailsMode.jsx

```diff
@@ -11,10 +10,8 @@ class PipelineStepDetailsMode extends React.Component {
       return null;
     }
 
-    const fromStepNameToFile = groupBy("fromStepName", inputFiles);
-    const fileGroupList = Object.keys(fromStepNameToFile).map(fromStepName => {
-      const fileGroup = fromStepNameToFile[fromStepName];
-      const fileList = fileGroup.map((file, i) => {
+    const fileGroupList = inputFiles.map((inputFileGroup, i) => {
+      const fileList = inputFileGroup.files.map((file, i) => {
```

Comment:
Is `.files` guaranteed to be defined?

Reply:
This should be handled with isRequired in propTypes, right?

Reply:
Yes, saw you did it below.